frappe/database/mariadb/database.py

```diff
@@ -114,18 +114,23 @@ def set_execution_timeout(self, seconds: int):
 		self.sql("set session max_statement_time = %s", int(seconds))
 
 	def get_connection_settings(self) -> dict:
+		assert (
+			self.password
+		), "without unix domain socket connection, mariadb db password is currently reqired"
 		conn_settings = {
-			"host": self.host,
 			"user": self.user,
-			"password": self.password,
 			"conv": self.CONVERSION_MAP,
 			"charset": "utf8mb4",
 			"use_unicode": True,
 		}
 
 		if self.user not in (frappe.flags.root_login, "root"):
-			conn_settings["database"] = self.user
+			conn_settings["database"] = self.cur_db_name
 
+		if self.host:
+			conn_settings["host"] = self.host
+		if self.password:
+			conn_settings["password"] = self.password
```

Comment:
host and password are not strictly mandatory for a mariadb connection (e.g. when using `MYSQL_UNIX_PORT` and when `whoami` is accordingly setup in the database.

This change anticipates this variant.

Reply:
Correction, `pymysql` doesn't actually do that but a more faithful lib implementation probably would.